charts/argo/templates/workflow-controller-clusterrole.yaml

```diff
@@ -78,4 +78,15 @@ rules:
   verbs:
   - get
   - list
+{{- if .Values.controller.persistence }}
+- apiGroups:
+  - ""
+  resources:
+  - secrets
```

Comment:
It's imprudent to give the controller a permission on all the secrets in the namespace. If you choose to give permissions, you should target them to the secret names provided in `.Values.controller.persistence.mysql.userNameSecret`, `.Values.controller.persistence.mysql.passwordSecret`, `.Values.controller.persistence.postgresql.userNameSecret`, and `.Values.controller.persistence.postgresql.passwordSecret`.

Reply:
You are right 👍